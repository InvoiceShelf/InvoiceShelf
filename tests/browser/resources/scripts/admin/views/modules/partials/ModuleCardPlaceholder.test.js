import { expect, test } from 'vitest'
import { render } from 'vitest-browser-vue'
import BaseContentPlaceholders from '@/scripts/components/base/BaseContentPlaceholders.vue'
import BaseContentPlaceholdersText from '@/scripts/components/base/BaseContentPlaceholdersText.vue'
import BaseContentPlaceholdersBox from '@/scripts/components/base/BaseContentPlaceholdersBox.vue'
import ModuleCardPlaceholder from '@/scripts/admin/views/modules/partials/ModuleCardPlaceholder.vue'

test('renders BaseContentPlaceholders with BaseContentPlaceholdersText using render function', async () => {
  const { getByTestId } = render(ModuleCardPlaceholder, {
    props: {
      'data-testid': 'modal-placeholder'
    },
    global: {
      components: {
        BaseContentPlaceholders,
        BaseContentPlaceholdersText,
        BaseContentPlaceholdersBox,
      },
    },
  })
  const modalPlaceholder = getByTestId('modal-placeholder')
  await expect.element(modalPlaceholder).toBeInTheDocument()
  expect(modalPlaceholder).toMatchSnapshot()
})


