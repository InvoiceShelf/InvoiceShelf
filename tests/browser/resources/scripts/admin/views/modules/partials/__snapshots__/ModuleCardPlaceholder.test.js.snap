// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders BaseContentPlaceholders with BaseContentPlaceholdersText using render function 1`] = `
PlaywrightLocator {
  "_container": <body>
    
    
    
  


    <div>
      <div
        class="base-content-placeholders base-content-placeholders-is-animated"
        data-testid="modal-placeholder"
      >
        
        <div
          class="shadow-md border-2 border-gray-200 border-opacity-60 rounded-lg cursor-pointer overflow-hidden h-100"
        >
          <div
            class="base-content-placeholders-box base-content-placeholders-is-rounded h-48 lg:h-64 md:h-48 w-full"
          />
          <div
            class="px-6 py-5 flex flex-col bg-gray-50 flex-1 justify-between"
          >
            <div
              class="base-content-placeholders-text base-content-placeholders-is-rounded w-32 h-8"
            >
              
              <div
                class="w-full h-full base-content-placeholders-text__line"
              />
              
            </div>
            <div
              class="flex items-center mt-2"
            >
              <div
                class="base-content-placeholders-box h-10 w-10 rounded-full sm:inline-block mr-2"
              />
              <div>
                <div
                  class="base-content-placeholders-text base-content-placeholders-is-rounded w-32 h-8 ml-2"
                >
                  
                  <div
                    class="w-full h-full base-content-placeholders-text__line"
                  />
                  
                </div>
              </div>
            </div>
            <div
              class="base-content-placeholders-text base-content-placeholders-is-rounded pt-4 w-full h-16"
            >
              
              <div
                class="w-full h-full base-content-placeholders-text__line"
              />
              
            </div>
            <div
              class="flex justify-between mt-4 flex-col space-y-2 sm:space-y-0 sm:flex-row"
            >
              <div
                class="base-content-placeholders-text base-content-placeholders-is-rounded w-32 h-8"
              >
                
                <div
                  class="w-full h-full base-content-placeholders-text__line"
                />
                
              </div>
              <div
                class="base-content-placeholders-text base-content-placeholders-is-rounded w-32 h-8"
              >
                
                <div
                  class="w-full h-full base-content-placeholders-text__line"
                />
                
              </div>
            </div>
          </div>
        </div>
        
      </div>
    </div>
  </body>,
  "_parsedSelector": {
    "capture": undefined,
    "parts": [
      {
        "body": [
          {
            "simples": [
              {
                "combinator": "",
                "selector": {
                  "css": "body",
                  "functions": [],
                },
              },
            ],
          },
        ],
        "name": "css",
        "source": "body",
      },
      {
        "body": "[data-testid="modal-placeholder"s]",
        "name": "internal:testid",
        "source": "[data-testid="modal-placeholder"s]",
      },
    ],
  },
  "_pwSelector": undefined,
  "selector": "body >> internal:testid=[data-testid="modal-placeholder"s]",
}
`;
